src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -197,7 +197,7 @@ public class ShadeMojo extends AbstractMojo {
      * <p/>
      * If you like to change the name of the native artifact, you may use the &lt;build>&lt;finalName> setting. If this
      * is set to something different than &lt;build>&lt;finalName>, no file replacement will be performed, even if
-     * shadedArtifactAttached is being used.
+     * {@link #shadedArtifactAttached} is false.
```

Comment:
Is this change correct? If I understood it correctly, if `<build><finalName>` and the Mojo `<finalName>` differ then the main project artifact will not be replaced, even if `shadedArtifactAttached=false`.

Reply:
to me, "shadedArtifactAttached is being used" means setting is t to `true`

Reply:
this part of the PR should IMHO just be dropped

Reply:
Ok, I have reverted that change.